src/core.js

```diff
@@ -1159,7 +1160,7 @@ define([
     };
 
     fn.loadXML = function (formXML, options) {
-        var form, _this = this;
+        var form, _this = this, qId = window.location.hash.substr(1);
```

Comment:
What do you think of dropping the extra `#` from the URL? So it would become commcarehq.org/a/..../source/#form/question1 Then if `qId` is a non-empty string and it starts with `"form/"` then prepend `#` and load that question else ignore `qId`.